apis/iam/v1beta1/diff_test.go

```diff
@@ -0,0 +1,856 @@
+					{
+						ID:  "spec.auditConfigs[service=compute.googleapis.com]",
+						Old: "absent in desired spec",
+						New: "present in actual spec",
+					},
+				},
+			}},
+		{
+			name: "audit config log type differs",
+			desired: &IAMPolicySpec{
+				AuditConfigs: []IAMPolicyAuditConfig{
+					{
+						Service: "allServices",
+						AuditLogConfigs: []AuditLogConfig{
+							{
+								LogType: "DATA_READ",
+							},
+						},
+					},
+				},
+			},
+			actual: &IAMPolicySpec{
+				AuditConfigs: []IAMPolicyAuditConfig{
+					{
+						Service: "allServices",
+						AuditLogConfigs: []AuditLogConfig{
+							{
+								LogType: "DATA_WRITE",
+							},
+						},
+					},
+				},
+			},
+			want: &structuredreporting.Diff{
+				Fields: []structuredreporting.DiffField{
+					{
+						ID: "spec.auditConfigs[service=allServices]",
+						Old: IAMPolicyAuditConfig{
+							Service: "allServices",
+							AuditLogConfigs: []AuditLogConfig{
+								{
+									LogType: "DATA_READ",
+								},
+							},
+						},
+						New: IAMPolicyAuditConfig{
+							Service: "allServices",
+							AuditLogConfigs: []AuditLogConfig{
+								{
+									LogType: "DATA_WRITE",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "audit config exempted members differs",
+			desired: &IAMPolicySpec{
+				AuditConfigs: []IAMPolicyAuditConfig{
+					{
+						Service: "allServices",
+						AuditLogConfigs: []AuditLogConfig{
+							{
+								LogType:         "DATA_READ",
+								ExemptedMembers: []Member{"user:exempt1@example.com"},
+							},
+						},
+					},
+				},
+			},
+			actual: &IAMPolicySpec{
+				AuditConfigs: []IAMPolicyAuditConfig{
+					{
+						Service: "allServices",
+						AuditLogConfigs: []AuditLogConfig{
+							{
+								LogType:         "DATA_READ",
+								ExemptedMembers: []Member{"user:exempt2@example.com"},
+							},
+						},
+					},
+				},
+			},
+			want: &structuredreporting.Diff{
+				Fields: []structuredreporting.DiffField{
+					{
+						ID: "spec.auditConfigs[service=allServices]",
+						Old: IAMPolicyAuditConfig{
+							Service: "allServices",
+							AuditLogConfigs: []AuditLogConfig{
+								{
+									LogType:         "DATA_READ",
+									ExemptedMembers: []Member{"user:exempt1@example.com"},
+								},
+							},
+						},
+						New: IAMPolicyAuditConfig{
+							Service: "allServices",
+							AuditLogConfigs: []AuditLogConfig{
+								{
+									LogType:         "DATA_READ",
+									ExemptedMembers: []Member{"user:exempt2@example.com"},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "audit config exempted members order differs - no diff",
+			desired: &IAMPolicySpec{
+				AuditConfigs: []IAMPolicyAuditConfig{
+					{
+						Service: "allServices",
+						AuditLogConfigs: []AuditLogConfig{
+							{
+								LogType:         "DATA_READ",
+								ExemptedMembers: []Member{"user:exempt1@example.com", "user:exempt2@example.com"},
```

Comment:
Paranoia. Your example has desired with ExemptedMembers in alphabetic order and actual with ExemptedMembers in NOT alphabetic order. Please add the inverse test.

So desired with ExemptedMembers in NOT alphabetic order and actual with ExemptedMembers in alphabetic order.